### PR TITLE
Update license to new terms

### LINKED ISSUE #42: Relicense documentation to creative commons
We currently have the website code under MIT license, but the MIT license is generally used for code. For the actual writing content, it would probably be better to use a license like CC-BY-4.0. Since the website is at a very early stage, I'm thinking that it would be easier to make this license change now rather than later.

In particular, I propose:
* The existing website code, configuration, and supporting technical files are licensed under MIT.
* Written text files (e.g. markdown files) and images are licensed under some creative commons license.

This license change would require us to get permission from the past contributors.

Thoughts? @rolandlo

### PR DESCRIPTION
**Pull request by Technius**
Fixes #42

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,3 +1,13 @@
+The website code, configuration, and supporting technical files are licensed
+under the MIT License.
+
+The written text content (e.g., markdown files) and images are licensed under
+CC-BY-SA 4.0.
```
Comment (rolandlo) [inline LICENSE:5]:
```suggestion
CC BY-SA 4.0.
```
(compare https://creativecommons.org/licenses/by-sa/4.0/)

**Review by rolandlo — COMMENTED** [review]
I have only found one flaw (the hyphen in CC BY-SA 4.0). Otherwise LGTM.

File: LICENSE [hunk 2]
```diff
@@ -1,3 +1,13 @@
+The website code, configuration, and supporting technical files are licensed
```
Comment (LittleHuba) [inline LICENSE:1]:
```suggestion
The content of this repository is licensed
```

Keep it as unrestrictive as possible. We don't want to only cover part of the stuff via this license but everything in this repository.

**Review by LittleHuba — CHANGES_REQUESTED** [review]
Looks good except for the limitation to specific content of the repository.

**Comment by LittleHuba** [comment]
I think it is problematic that we have two separate licenses for the webpage. In my opinion, it is an either-or decision. We should go with CC BY-SA 4.0 for everything.

**Comment by Technius** [comment]
In general, moving to CC is _more restrictive_ than staying with MIT license. This repository still contains a sizable amount of code, which [CC is not really designed to handle](https://opensource.stackexchange.com/questions/1717/why-is-cc-by-sa-discouraged-for-code). The people who are copying/citing the written text will most likely be separate from the people who are copying the code/configuration for use in their own mkdocs setups. CC is quite restrictive for the latter crowd, so I believe we should be treating these crowds differently in terms of license.

**Comment by LittleHuba** [comment]
Hmm good point. Though for me the situation becomes really awful if we have files that contain both code and text. Those files are not clearly associated with either license. That is a real mess if you want to reuse anything. If we do this, files must be strictly separated and code must be contained within its own directory. Otherwise it is not clear what is covered by what license.

**Comment by Technius** [comment]
> Though for me the situation becomes really awful if we have files that contain both code and text.

I agree with this, but I don't think this is a very big issue. Most the situations that require code are related to page layout, so we can easily split them into layout code in templates + written text in markdown files. For example, see how the installation page is split into [layout](https://github.com/xournalpp/xournalpp.github.io/tree/site/overrides/installation) and [textual content](https://github.com/xournalpp/xournalpp.github.io/tree/site/docs/installation).

On that note, the home page files could use some refactoring.

**Comment by Technius** [comment]
Merging in 24 hours unless there are further objections.

**Comment by Technius** [comment]
Sorry for the delay, was busy this week. Merging now.

**Pull request merged**